File: back/src/main/java/com/back/domain/user/dto/SignupResponse.java
package com.back.domain.user.dto;

import com.back.domain.user.entity.User;

public record SignupResponse(
        Long userId,
        String loginType
) {
    public static SignupResponse from(User user) {
        return new SignupResponse(user.getUserId(), user.getLoginType().name());
    }
}

File: back/src/main/kotlin/com/back/domain/user/dto/SignupResponse.kt
package com.back.domain.user.dto

import com.back.domain.user.entity.User

data class SignupResponse(
    val userId: Long,
    val loginType: String
) {
    companion object {
        fun from(user: User): SignupResponse {
            val userId = user.userId ?: throw IllegalArgumentException("User ID must not be null")
            return SignupResponse(userId, user.loginType.name)
        }
    }
}
